From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Greg Kroah-Hartman <gregkh@google.com>
Date: Wed, 29 Apr 2020 15:20:54 +0200
Subject: ANDROID: cfi: fix export symbol types

In commit d71a92631c51 ("ANDROID: add support for Clang's Control Flow
Integrity (CFI)") some new symbols were exported, but they should have
been set as _GPL symbols.

Fix this up by properly.

Bug: 145210207
Cc: Sami Tolvanen <samitolvanen@google.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@google.com>
Change-Id: I6ecbb0f3b33f7c02c9b75bb7d80c35ce80e553f3
---
 kernel/cfi.c | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/kernel/cfi.c b/kernel/cfi.c
--- a/kernel/cfi.c
+++ b/kernel/cfi.c
@@ -216,14 +216,14 @@ void cfi_module_add(struct module *mod, unsigned long min_addr,
 {
 	update_shadow(mod, min_addr, max_addr, add_module_to_shadow);
 }
-EXPORT_SYMBOL(cfi_module_add);
+EXPORT_SYMBOL_GPL(cfi_module_add);
 
 void cfi_module_remove(struct module *mod, unsigned long min_addr,
 	unsigned long max_addr)
 {
 	update_shadow(mod, min_addr, max_addr, remove_module_from_shadow);
 }
-EXPORT_SYMBOL(cfi_module_remove);
+EXPORT_SYMBOL_GPL(cfi_module_remove);
 
 static inline cfi_check_fn ptr_to_check_fn(const struct cfi_shadow __rcu *s,
 	unsigned long ptr)
@@ -292,14 +292,14 @@ void cfi_slowpath_handler(uint64_t id, void *ptr, void *diag)
 	else /* Don't allow unchecked modules */
 		handle_cfi_failure(ptr);
 }
-EXPORT_SYMBOL(cfi_slowpath_handler);
+EXPORT_SYMBOL_GPL(cfi_slowpath_handler);
 #endif /* CONFIG_MODULES */
 
 void cfi_failure_handler(void *data, void *ptr, void *vtable)
 {
 	handle_cfi_failure(ptr);
 }
-EXPORT_SYMBOL(cfi_failure_handler);
+EXPORT_SYMBOL_GPL(cfi_failure_handler);
 
 void __cfi_check_fail(void *data, void *ptr)
 {
